gnovm/pkg/gnolang/values_string.go

```diff
@@ -180,66 +181,24 @@ func (tv *TypedValue) Sprint(m *Machine) string {
 	if _, ok := tv.T.(*DeclaredType); ok {
 		return tv.String()
 	}
+
 	// otherwise, default behavior.
 	switch bt := baseOf(tv.T).(type) {
 	case PrimitiveType:
-		switch bt {
-		case UntypedBoolType, BoolType:
-			return fmt.Sprintf("%t", tv.GetBool())
-		case UntypedStringType, StringType:
-			return tv.GetString()
-		case IntType:
-			return fmt.Sprintf("%d", tv.GetInt())
-		case Int8Type:
-			return fmt.Sprintf("%d", tv.GetInt8())
-		case Int16Type:
-			return fmt.Sprintf("%d", tv.GetInt16())
-		case UntypedRuneType, Int32Type:
-			return fmt.Sprintf("%d", tv.GetInt32())
-		case Int64Type:
-			return fmt.Sprintf("%d", tv.GetInt64())
-		case UintType:
-			return fmt.Sprintf("%d", tv.GetUint())
-		case Uint8Type:
-			return fmt.Sprintf("%d", tv.GetUint8())
-		case Uint16Type:
-			return fmt.Sprintf("%d", tv.GetUint16())
-		case Uint32Type:
-			return fmt.Sprintf("%d", tv.GetUint32())
-		case Uint64Type:
-			return fmt.Sprintf("%d", tv.GetUint64())
-		case Float32Type:
-			return fmt.Sprintf("%v", tv.GetFloat32())
-		case Float64Type:
-			return fmt.Sprintf("%v", tv.GetFloat64())
-		case UntypedBigintType, BigintType:
-			return tv.V.(BigintValue).V.String()
-		case UntypedBigdecType, BigdecType:
-			return tv.V.(BigdecValue).V.String()
-		default:
-			panic("should not happen")
-		}
+		return printPrimitive(bt, *tv)
```

Comment:
I think it is easier to read if all the types are in the same switch statement like it was, both primitive and others.  Now I have to navigate to another function that is only called once from this location; it seems less convenient for anyone reading the code